detox/src/artifacts/log/ios/SimulatorLogRecording.js

```diff
@@ -0,0 +1,81 @@
+const fs = require('fs-extra');
+const npmlog = require('npmlog');
+const { Tail } = require('tail');
+const ensureMove = require('../../../utils/ensureMove');
+const RecordingArtifact = require('../../core/artifact/RecordingArtifact');
+
+class SimulatorLogRecording extends RecordingArtifact {
+  constructor({
+    stdoutPath,
+    stderrPath,
+    temporaryLogPath,
+    fromBeginning,
+  }) {
+    super();
+
+    this._logPath = temporaryLogPath;
+    this._stdoutPath = stdoutPath;
+    this._stderrPath = stderrPath;
+    this._fromBeginning = fromBeginning;
+
+    this._logStream = null;
+    this._stdoutTail = null;
+    this._stderrTail = null;
+  }
+
+  async doStart() {
+    await fs.ensureFile(this._logPath);
+    this._logStream = fs.createWriteStream(this._logPath, { flags: 'w' });
+    this._stdoutTail = this._createTail(this._stdoutPath, 'stdout');
+    this._stderrTail = this._createTail(this._stderrPath, 'stderr');
+  }
+
+  async doStop() {
+    this._stdoutTail.unwatch();
+    this._stderrTail.unwatch();
+    this._logStream.end();
+  }
+
+  async doSave(artifactPath) {
+    await ensureMove(this._logPath, artifactPath, '.log');
+  }
+
+  async doDiscard() {
+    await fs.remove(this._logPath);
+  }
+
+  _createTail(file, prefix) {
+    const tail = new Tail(file, {
```

Comment:
I'm afraid tailing those files might be CPU intensive, let's think together if there's something out of process that we can do to avoid reading line by line.
Previously we have redirected std to files, and collected the files after each test. It's ugly, I agree, but much more efficient.